File: 0001-two-sum/solution.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long ll;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        auto arr=nums;
        vector<int>ans;
        ll len=arr.size();
        map<ll,ll>mp;
        for(ll i=0;i<len;i++){
            ll val=target-arr[i];
            if(mp[val]){
                ans.push_back(mp[val]-1);
                ans.push_back(i);
                return ans;
            }else mp[arr[i]]=i+1;
        }
        return ans;
    }
};

File: 0001-two-sum/solution.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int[] ans = new int[2];
        Dictionary<int,int> dic = new Dictionary<int,int>();
        int len = nums.Length;
        for(int i=0;i<len;i++){
            int rem = target - nums[i];
            if(dic.ContainsKey(rem)){
                ans[0]=dic[rem];
                ans[1]=i;
                return ans;
            }
            dic[nums[i]]=i;
        }
        return ans;
    }
}
